### PR TITLE
[STRAT-53] Show energy level over the course of a race

### PR DESCRIPTION
**Pull request by e-wai**
Creates a graph of state of charge and velocity against distance travelled. Add functions to generate desired SOC values given velocity and elevation profiles. Uses Car and ColoumbCounter objects to calculate energy used and state of charge.

### CODE REVIEW

**Comment by clarkevandenhoven** [comment]
> Should I write tests for either of the functions? Not sure how to test the graphs but I can probably test that calculateSOCValues works as expected?

You don't need to test graph generation, but please add a test for Calculating SOC Values

File: displays/SOC_velocity_graph.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    ax2.set_ylabel('State of Charge (%)')
+    ax2.yaxis.label.set_color(color)
+    ax2.spines['right'].set_color(color)
+    ax2.plot(x_axis, soc_percentages, color=color)
+
+    plt.title('State of Charge vs. Velocity Graph')
+    fig.tight_layout()
+    plt.show()
+
+velocities = [5, 7, 9, 13, 9.5, 7.5, 5]
```
Comment (clarkevandenhoven) [inline displays/SOC_velocity_graph.py:63]: This should be wrapped into a test. The way to do this is probably to have the graph saved in a file, and check that the byte stream of the file is equivalent to the one you are generating with the function. Ping me if you need help

File: displays/SOC_velocity_graph.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    car_model = Car()
+    soc_est = CoulombCounter()
+
+    soc_intervals = [initial_soc]
+    soc_est.set_SoC(initial_soc)
+
+    for i in range(len(v_profile) - 1):
+        energy_used = car_model.energy_used(v_profile[i: i + 2], e_profile[i: i + 2], distance[i: i +1])
+        #in SoC calculation, wanted units is W = Js^-1, cancelled out in function's conversion so using 1 
+        soc_est.discharge(energy_used / 1, 1, True)
```
[the remark this answers is not in the document]
Comment (e-wai) [reply displays/SOC_velocity_graph.py:19]: discharge() takes value in W and converts it to kWh but energy_used result is given in J. I believe the direct conversion for J to kWh is /3600000. for the conversion in discharge(), it multiplies by time before dividing by 3600000, so as long as the denom and time passed match it shouldn't matter. 

TL;DR i divided by 1 for completion sake since it is asking for W but it doesn't do anything... I can delete it if you want

File: displays/SOC_velocity_graph.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+def calculateSOCValues(v_profile, e_profile, distance, initial_soc):
+    car_model = Car()
+    soc_est = CoulombCounter()
+
+    soc_intervals = [initial_soc]
+    soc_est.set_SoC(initial_soc)
+
+    for i in range(len(v_profile) - 1):
+        energy_used = car_model.energy_used(v_profile[i: i + 2], e_profile[i: i + 2], distance[i: i +1])
```
[the remark this answers is not in the document]
Comment (e-wai) [reply displays/SOC_velocity_graph.py:17]: should've asked about this one: energy_used calculates energy used on interval of two velocities. should i assume it remains constant over the interval and not consider energy to get to next velocity?

[the remark this answers is not in the document]
Comment (clarkevandenhoven) [reply displays/SOC_velocity_graph.py:17]: No this is the smarter way to do it :)
